src/app/drive/components/DriveExplorer/DriveExplorer.tsx

```diff
@@ -458,31 +466,53 @@ declare module 'react' {
 
 const uploadItems = async (props: DriveExplorerProps, rootList: IRoot[], files: File[]) => {
   const { dispatch, currentFolderId, onDragAndDropEnd } = props;
-  if (files.length) {
-    // files where dragged directly
-    await dispatch(
-      storageThunks.uploadItemsThunkNoCheck({
-        files,
-        parentFolderId: currentFolderId,
-        options: {
-          onSuccess: onDragAndDropEnd,
-        },
-      }),
-    );
-  }
 
-  if (rootList.length) {
-    for (const root of rootList) {
+  let totalFilesToUpload: number;
+
+  totalFilesToUpload = files.length;
+
+  const countTotalItemsToUpload = (rootList) => {
+    rootList.forEach((n) => {
+      totalFilesToUpload += n.childrenFiles.length;
+      if (n.childrenFolders.length >= 1) {
+        countTotalItemsToUpload(n.childrenFolders);
+      }
+    });
+  };
+
+  countTotalItemsToUpload(rootList);
+
+  if (totalFilesToUpload < 1000) {
+    if (files.length) {
+      // files where dragged directly
       await dispatch(
-        storageThunks.uploadFolderThunkNoCheck({
-          root,
-          currentFolderId,
+        storageThunks.uploadItemsThunkNoCheck({
+          files,
+          parentFolderId: currentFolderId,
           options: {
             onSuccess: onDragAndDropEnd,
           },
         }),
       );
     }
+    if (rootList.length) {
+      for (const root of rootList) {
+        await dispatch(
+          storageThunks.uploadFolderThunkNoCheck({
+            root,
+            currentFolderId,
+            options: {
+              onSuccess: onDragAndDropEnd,
+            },
+          }),
+        );
+      }
+    }
+  } else {
+    notificationsService.show({
+      text: 'The maximum is 1000 files per upload.',
```

Comment:
I think this warning is ok, although as an opinion, we might need a bigger warning sign.

Please check with Alvaro if we can have a prompt with a button of `ok` so that we are sure the user got the feedback >1000 files is too much. Instead of a vanishing one.

Reply:
added dialog component

Reply:
nice